lib/charms/karma_k8s/v0/karma.py

```diff
@@ -165,22 +174,20 @@ class KarmaConsumer(ConsumerBase):
     Arguments:
             charm (CharmBase): consumer charm
             name (str): from consumer's metadata.yaml
-            consumes (dict): provider specifications
-            multi (bool): multiple relations flag
 
     Attributes:
             charm (CharmBase): consumer charm
     """
 
     on = KarmaConsumerEvents()
 
-    def __init__(self, charm, name: str, consumes: dict, multi: bool = False):
-        super().__init__(charm, name, consumes, multi)
+    def __init__(self, charm, name: str):
+        super().__init__(charm, name)
         self.charm = charm
 
         events = self.charm.on[self.name]
-        self.framework.observe(events.relation_changed, self._on_relation_changed)
-        self.framework.observe(events.relation_departed, self._on_relation_departed)
+        self.charm.framework.observe(events.relation_changed, self._on_relation_changed)
```

Comment:
`Object` also has a `framework` instance. I am not entirely sure if this change is necessary.

Reply:
Leftover. Fixed.